### DISCUSSION TITLE
Detection of LICENSE containing only mandatory text

### DISCUSSION THREAD

**Discussion by metacubed** [community/community#65838] [Repositories] [answered]
### Select Topic Area

Product Feedback

### Body

Most common licenses have "How to apply" instructions in the Appendix section, which ask you to include just a short paragraph (example at the end of this post).

However, if you follow these instructions and include just the relevant section, GitHub is not able to detect the license on the repository.

Please update license detection to recognize this kind of usage. In addition, new repository generation could also be changed to use just the relevant section instead of the entire text.

For example, the Apache 2.0 license says:
>   APPENDIX: How to apply the Apache License to your work.
> 
>   To apply the Apache License to your work, attach the following
>   boilerplate notice, with the fields enclosed by brackets "[]"
>   replaced with your own identifying information. (Don't include
>   the brackets!)  The text should be enclosed in the appropriate
>   comment syntax for the file format. We also recommend that a
>   file or class name and description of purpose be included on the
>   same "printed page" as the copyright notice for easier
>   identification within third-party archives.

So, you would actually need only this content in your LICENSE file:
```
   Copyright 2023 Example Corp.

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
```

**Comment by rouke-broersma** [comment, marked as answer]
@metacubed I don't think this is correct. The APPENDIX is for applying the license information to your source files. You are required to provide a copy of the full license text with your project in the LICENSE file.
